[2023-05-29T16:29:50.862+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:871: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti: TaskInstance | None = qry.with_for_update().one_or_none()

[2023-05-29T16:29:50.933+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: word_count_1.word_count manual__2023-05-29T16:28:21.024578+00:00 [queued]>
[2023-05-29T16:29:50.972+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: word_count_1.word_count manual__2023-05-29T16:28:21.024578+00:00 [queued]>
[2023-05-29T16:29:50.973+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2023-05-29T16:29:50.973+0000] {taskinstance.py:1363} INFO - Starting attempt 2 of 2
[2023-05-29T16:29:50.974+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2023-05-29T16:29:51.033+0000] {taskinstance.py:1383} INFO - Executing <Task(SparkSubmitOperator): word_count> on 2023-05-29 16:28:21.024578+00:00
[2023-05-29T16:29:51.048+0000] {standard_task_runner.py:54} INFO - Started process 3439 to run task
[2023-05-29T16:29:51.061+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'word_count_1', 'word_count', 'manual__2023-05-29T16:28:21.024578+00:00', '--job-id', '4579', '--raw', '--subdir', 'DAGS_FOLDER/word_count_scheduler.py', '--cfg-path', '/tmp/tmpjksr4go_']
[2023-05-29T16:29:51.063+0000] {standard_task_runner.py:83} INFO - Job 4579: Subtask word_count
[2023-05-29T16:29:51.065+0000] {dagbag.py:525} INFO - Filling up the DagBag from /opt/***/dags/word_count_scheduler.py
[2023-05-29T16:29:51.347+0000] {task_command.py:384} INFO - Running <TaskInstance: word_count_1.word_count manual__2023-05-29T16:28:21.024578+00:00 [running]> on host fb138c374944
[2023-05-29T16:29:51.356+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:873: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti = qry.one_or_none()

[2023-05-29T16:29:51.653+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=pioneer22022001@gmail.com
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=word_count_1
AIRFLOW_CTX_TASK_ID=word_count
AIRFLOW_CTX_EXECUTION_DATE=2023-05-29T16:28:21.024578+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-05-29T16:28:21.024578+00:00
[2023-05-29T16:29:51.683+0000] {base.py:71} INFO - Using connection ID 'other_spark_local' for task execution.
[2023-05-29T16:29:51.686+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name word_count --verbose /opt/spark/app/word_count.py
[2023-05-29T16:29:51.729+0000] {spark_submit.py:490} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2023-05-29T16:29:54.731+0000] {spark_submit.py:490} INFO - Using properties file: null
[2023-05-29T16:29:55.136+0000] {spark_submit.py:490} INFO - Parsed arguments:
[2023-05-29T16:29:55.137+0000] {spark_submit.py:490} INFO - master                  spark://spark-master:7077
[2023-05-29T16:29:55.138+0000] {spark_submit.py:490} INFO - deployMode              null
[2023-05-29T16:29:55.138+0000] {spark_submit.py:490} INFO - executorMemory          null
[2023-05-29T16:29:55.139+0000] {spark_submit.py:490} INFO - executorCores           null
[2023-05-29T16:29:55.139+0000] {spark_submit.py:490} INFO - totalExecutorCores      null
[2023-05-29T16:29:55.139+0000] {spark_submit.py:490} INFO - propertiesFile          null
[2023-05-29T16:29:55.139+0000] {spark_submit.py:490} INFO - driverMemory            null
[2023-05-29T16:29:55.140+0000] {spark_submit.py:490} INFO - driverCores             null
[2023-05-29T16:29:55.140+0000] {spark_submit.py:490} INFO - driverExtraClassPath    null
[2023-05-29T16:29:55.141+0000] {spark_submit.py:490} INFO - driverExtraLibraryPath  null
[2023-05-29T16:29:55.141+0000] {spark_submit.py:490} INFO - driverExtraJavaOptions  null
[2023-05-29T16:29:55.141+0000] {spark_submit.py:490} INFO - supervise               false
[2023-05-29T16:29:55.141+0000] {spark_submit.py:490} INFO - queue                   null
[2023-05-29T16:29:55.142+0000] {spark_submit.py:490} INFO - numExecutors            null
[2023-05-29T16:29:55.142+0000] {spark_submit.py:490} INFO - files                   null
[2023-05-29T16:29:55.142+0000] {spark_submit.py:490} INFO - pyFiles                 null
[2023-05-29T16:29:55.142+0000] {spark_submit.py:490} INFO - archives                null
[2023-05-29T16:29:55.143+0000] {spark_submit.py:490} INFO - mainClass               null
[2023-05-29T16:29:55.143+0000] {spark_submit.py:490} INFO - primaryResource         file:/opt/spark/app/word_count.py
[2023-05-29T16:29:55.143+0000] {spark_submit.py:490} INFO - name                    word_count
[2023-05-29T16:29:55.144+0000] {spark_submit.py:490} INFO - childArgs               []
[2023-05-29T16:29:55.144+0000] {spark_submit.py:490} INFO - jars                    null
[2023-05-29T16:29:55.145+0000] {spark_submit.py:490} INFO - packages                null
[2023-05-29T16:29:55.146+0000] {spark_submit.py:490} INFO - packagesExclusions      null
[2023-05-29T16:29:55.146+0000] {spark_submit.py:490} INFO - repositories            null
[2023-05-29T16:29:55.147+0000] {spark_submit.py:490} INFO - verbose                 true
[2023-05-29T16:29:55.147+0000] {spark_submit.py:490} INFO - 
[2023-05-29T16:29:55.148+0000] {spark_submit.py:490} INFO - Spark properties used, including those specified through
[2023-05-29T16:29:55.148+0000] {spark_submit.py:490} INFO - --conf and those from the properties file null:
[2023-05-29T16:29:55.149+0000] {spark_submit.py:490} INFO - (spark.master,spark://spark-master:7077)
[2023-05-29T16:29:55.150+0000] {spark_submit.py:490} INFO - 
[2023-05-29T16:29:55.150+0000] {spark_submit.py:490} INFO - 
[2023-05-29T16:29:56.132+0000] {spark_submit.py:490} INFO - Main class:
[2023-05-29T16:29:56.133+0000] {spark_submit.py:490} INFO - org.apache.spark.deploy.PythonRunner
[2023-05-29T16:29:56.136+0000] {spark_submit.py:490} INFO - Arguments:
[2023-05-29T16:29:56.137+0000] {spark_submit.py:490} INFO - file:/opt/spark/app/word_count.py
[2023-05-29T16:29:56.137+0000] {spark_submit.py:490} INFO - null
[2023-05-29T16:29:56.172+0000] {spark_submit.py:490} INFO - Spark config:
[2023-05-29T16:29:56.173+0000] {spark_submit.py:490} INFO - (spark.master,spark://spark-master:7077)
[2023-05-29T16:29:56.174+0000] {spark_submit.py:490} INFO - (spark.submit.pyFiles,)
[2023-05-29T16:29:56.174+0000] {spark_submit.py:490} INFO - (spark.app.name,word_count)
[2023-05-29T16:29:56.175+0000] {spark_submit.py:490} INFO - (spark.submit.deployMode,client)
[2023-05-29T16:29:56.181+0000] {spark_submit.py:490} INFO - Classpath elements:
[2023-05-29T16:29:56.182+0000] {spark_submit.py:490} INFO - 
[2023-05-29T16:29:56.182+0000] {spark_submit.py:490} INFO - 
[2023-05-29T16:29:56.183+0000] {spark_submit.py:490} INFO - 
[2023-05-29T16:29:56.240+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:873: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti = qry.one_or_none()

[2023-05-29T16:29:57.370+0000] {spark_submit.py:490} INFO - Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
[2023-05-29T16:29:57.397+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:57 INFO SparkContext: Running Spark version 3.2.3
[2023-05-29T16:29:57.781+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:57 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2023-05-29T16:29:58.200+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO ResourceUtils: ==============================================================
[2023-05-29T16:29:58.202+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO ResourceUtils: No custom resources configured for spark.driver.
[2023-05-29T16:29:58.203+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO ResourceUtils: ==============================================================
[2023-05-29T16:29:58.205+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO SparkContext: Submitted application: Firstprogram
[2023-05-29T16:29:58.309+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2023-05-29T16:29:58.381+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO ResourceProfile: Limiting resource is cpu
[2023-05-29T16:29:58.384+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2023-05-29T16:29:58.569+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO SecurityManager: Changing view acls to: ***
[2023-05-29T16:29:58.573+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO SecurityManager: Changing modify acls to: ***
[2023-05-29T16:29:58.576+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO SecurityManager: Changing view acls groups to:
[2023-05-29T16:29:58.578+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO SecurityManager: Changing modify acls groups to:
[2023-05-29T16:29:58.580+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:58 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2023-05-29T16:29:59.532+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO Utils: Successfully started service 'sparkDriver' on port 43509.
[2023-05-29T16:29:59.645+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO SparkEnv: Registering MapOutputTracker
[2023-05-29T16:29:59.769+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO SparkEnv: Registering BlockManagerMaster
[2023-05-29T16:29:59.845+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2023-05-29T16:29:59.848+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2023-05-29T16:29:59.862+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2023-05-29T16:29:59.944+0000] {spark_submit.py:490} INFO - 23/05/29 16:29:59 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-a25ec54c-d898-4abf-b6cf-fd119e17f2ae
[2023-05-29T16:30:00.025+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:00 INFO MemoryStore: MemoryStore started with capacity 366.3 MiB
[2023-05-29T16:30:00.083+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:00 INFO SparkEnv: Registering OutputCommitCoordinator
[2023-05-29T16:30:01.004+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:01 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2023-05-29T16:30:01.257+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:01 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://fb138c374944:4040
[2023-05-29T16:30:02.510+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:02 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2023-05-29T16:30:02.822+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:02 INFO TransportClientFactory: Successfully created connection to spark-master/172.27.0.2:7077 after 198 ms (0 ms spent in bootstraps)
[2023-05-29T16:30:03.310+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20230529163003-0005
[2023-05-29T16:30:03.333+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20230529163003-0005/0 on worker-20230529151449-172.27.0.3-7000 (172.27.0.3:7000) with 1 core(s)
[2023-05-29T16:30:03.343+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO StandaloneSchedulerBackend: Granted executor ID app-20230529163003-0005/0 on hostPort 172.27.0.3:7000 with 1 core(s), 1024.0 MiB RAM
[2023-05-29T16:30:03.391+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 37629.
[2023-05-29T16:30:03.392+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO NettyBlockTransferService: Server created on fb138c374944:37629
[2023-05-29T16:30:03.418+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2023-05-29T16:30:03.464+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20230529163003-0005/0 is now RUNNING
[2023-05-29T16:30:03.480+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, fb138c374944, 37629, None)
[2023-05-29T16:30:03.512+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO BlockManagerMasterEndpoint: Registering block manager fb138c374944:37629 with 366.3 MiB RAM, BlockManagerId(driver, fb138c374944, 37629, None)
[2023-05-29T16:30:03.565+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, fb138c374944, 37629, None)
[2023-05-29T16:30:03.572+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:03 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, fb138c374944, 37629, None)
[2023-05-29T16:30:04.641+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:04 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2023-05-29T16:30:06.655+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:06 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2023-05-29T16:30:06.829+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:06 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2023-05-29T16:30:13.278+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:13 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 354.4 KiB, free 366.0 MiB)
[2023-05-29T16:30:13.545+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:13 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 32.0 KiB, free 365.9 MiB)
[2023-05-29T16:30:13.557+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:13 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on fb138c374944:37629 (size: 32.0 KiB, free: 366.3 MiB)
[2023-05-29T16:30:13.585+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:13 INFO SparkContext: Created broadcast 0 from textFile at NativeMethodAccessorImpl.java:0
[2023-05-29T16:30:13.953+0000] {spark_submit.py:490} INFO - Traceback (most recent call last):
[2023-05-29T16:30:13.954+0000] {spark_submit.py:490} INFO - File "/opt/spark/app/word_count.py", line 17, in <module>
[2023-05-29T16:30:13.956+0000] {spark_submit.py:490} INFO - # clean_text_file = text_file.map(lambda line: regexp_replace(line, '[^\w\s]', ''))
[2023-05-29T16:30:13.957+0000] {spark_submit.py:490} INFO - File "/usr/local/lib/python3.7/site-packages/pyspark/context.py", line 564, in parallelize
[2023-05-29T16:30:13.957+0000] {spark_submit.py:490} INFO - c = list(c)    # Make it a list so we can compute its length
[2023-05-29T16:30:13.958+0000] {spark_submit.py:490} INFO - TypeError: 'RDD' object is not iterable
[2023-05-29T16:30:14.096+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO SparkContext: Invoking stop() from shutdown hook
[2023-05-29T16:30:14.136+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO SparkUI: Stopped Spark web UI at http://fb138c374944:4040
[2023-05-29T16:30:14.155+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO StandaloneSchedulerBackend: Shutting down all executors
[2023-05-29T16:30:14.159+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2023-05-29T16:30:14.221+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2023-05-29T16:30:14.264+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO MemoryStore: MemoryStore cleared
[2023-05-29T16:30:14.265+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO BlockManager: BlockManager stopped
[2023-05-29T16:30:14.293+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO BlockManagerMaster: BlockManagerMaster stopped
[2023-05-29T16:30:14.301+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2023-05-29T16:30:14.328+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO SparkContext: Successfully stopped SparkContext
[2023-05-29T16:30:14.329+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO ShutdownHookManager: Shutdown hook called
[2023-05-29T16:30:14.334+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO ShutdownHookManager: Deleting directory /tmp/spark-591ac4ee-8534-4dfd-9cad-acb9ec24d97e
[2023-05-29T16:30:14.344+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO ShutdownHookManager: Deleting directory /tmp/spark-591ac4ee-8534-4dfd-9cad-acb9ec24d97e/pyspark-c84ca867-d376-45e1-b472-a76ebe82b240
[2023-05-29T16:30:14.354+0000] {spark_submit.py:490} INFO - 23/05/29 16:30:14 INFO ShutdownHookManager: Deleting directory /tmp/spark-d10a664d-cb4f-4b03-a104-843312c51a15
[2023-05-29T16:30:14.552+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:871: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti: TaskInstance | None = qry.with_for_update().one_or_none()

[2023-05-29T16:30:14.576+0000] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name word_count --verbose /opt/spark/app/word_count.py. Error code is: 1.
[2023-05-29T16:30:14.586+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/models/taskinstance.py:873: SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  ti = qry.one_or_none()

[2023-05-29T16:30:14.604+0000] {taskinstance.py:1406} INFO - Marking task as FAILED. dag_id=word_count_1, task_id=word_count, execution_date=20230529T162821, start_date=20230529T162950, end_date=20230529T163014
[2023-05-29T16:30:14.661+0000] {warnings.py:110} WARNING - /usr/local/lib/python3.7/site-packages/***/utils/email.py:120: RemovedInAirflow3Warning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
  send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)

[2023-05-29T16:30:14.663+0000] {configuration.py:569} WARNING - section/key [smtp/smtp_user] not found in config
[2023-05-29T16:30:14.664+0000] {email.py:229} INFO - Email alerting: attempt 1
[2023-05-29T16:30:14.692+0000] {configuration.py:569} WARNING - section/key [smtp/smtp_user] not found in config
[2023-05-29T16:30:14.693+0000] {email.py:229} INFO - Email alerting: attempt 1
[2023-05-29T16:30:14.694+0000] {taskinstance.py:1914} ERROR - Failed to send email to: ['pioneer22022001@gmail.com']
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1457, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1603, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1664, in _execute_task
    result = execute_callable(context=context)
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/usr/local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name word_count --verbose /opt/spark/app/word_count.py. Error code is: 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2311, in email_alert
    send_email(task.email, subject, html_content)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 72, in send_email
    **kwargs,
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 120, in send_email_smtp
    send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 231, in send_mime_email
    smtp_conn = _get_smtp_connection(smtp_host, smtp_port, smtp_timeout, smtp_ssl)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 265, in _get_smtp_connection
    else smtplib.SMTP(host=host, port=port, timeout=timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.7/smtplib.py", line 336, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 307, in _get_socket
    self.source_address)
  File "/usr/local/lib/python3.7/socket.py", line 728, in create_connection
    raise err
  File "/usr/local/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1912, in handle_failure
    self.email_alert(error, task)
  File "/usr/local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2313, in email_alert
    send_email(task.email, subject, html_content_err)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 72, in send_email
    **kwargs,
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 120, in send_email_smtp
    send_mime_email(e_from=mail_from, e_to=recipients, mime_msg=msg, conn_id=conn_id, dryrun=dryrun)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 231, in send_mime_email
    smtp_conn = _get_smtp_connection(smtp_host, smtp_port, smtp_timeout, smtp_ssl)
  File "/usr/local/lib/python3.7/site-packages/airflow/utils/email.py", line 265, in _get_smtp_connection
    else smtplib.SMTP(host=host, port=port, timeout=timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 251, in __init__
    (code, msg) = self.connect(host, port)
  File "/usr/local/lib/python3.7/smtplib.py", line 336, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/usr/local/lib/python3.7/smtplib.py", line 307, in _get_socket
    self.source_address)
  File "/usr/local/lib/python3.7/socket.py", line 728, in create_connection
    raise err
  File "/usr/local/lib/python3.7/socket.py", line 716, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
[2023-05-29T16:30:14.756+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 4579 for task word_count (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name word_count --verbose /opt/spark/app/word_count.py. Error code is: 1.; 3439)
[2023-05-29T16:30:14.825+0000] {local_task_job.py:164} INFO - Task exited with return code 1
[2023-05-29T16:30:14.973+0000] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
